duckietown/dt-ros-commons
# Daffy new deal timing

## Description

Timing incorporated in DTROS.
Can be tested with the camera node from https://github.com/duckietown/dt-duckiebot-interface/tree/daffy-with-timing

To test:
1. In this repo: `dts devel build -f -H donald`
2. In `dt-duckiebot-interface:daffy-with-timing`: `dts devel build -f -b daffy-new-deal-timing -H donald`
3. Start the demo: `dts duckiebot demo -b donald -d camera_node -p camera_driver -i duckietown/dt-duckiebot-interface:daffy-with-timing-arm32v7`
4. Check out the timing msgs: `docker -H donald.local exec -it demo_camera_node bash -c "source ../../devel/setup.bash; rostopic echo /donald/camera_node/phase_times"`


## Type of change

Please mark the relevant options with X:

- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Tests

Please describe all tests ran and their outcome. It is highly recommended that you at least do the tests listed by default, even if you only changed the documentation.

- [ ] Made the Duckiebot move with keyboard control
- [ ] Visually checked the published camera stream
- [ ] Verified that the lane following demo works
- [ ] Verified that the indefinite navigation demo works

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [ ] I checked that this pull request is compliant with the Duckietown code style
- [ ] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [ ] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [ ] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [ ] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_
